Material dialog split

ivoberger opened the pull request:
As noted in the corresponding issue LouisCAD/Splitties#168 `MaterialDialogAlertBuilder` isn't available in a stable release of material components yet. I'll update this once it's out.
I'm opening this now so you can take a look.
Closes #168

LouisCAD commented on modules/materialdialog/src/androidMain/kotlin/splitties/materialdialog/MaterialDialog.kt:
```diff
@@ -0,0 +1,62 @@
+inline fun Context.materialDialog(
+    title: CharSequence? = null,
+    message: CharSequence? = null,
+    icon: Drawable?,
```
`icon` should default to `null`.

IIIuminator replied:
That might need to be changed in the `alertdialog` and `alertdialog-appcompat`too as I just copied the code and changed the builder

LouisCAD replied:
These splits have overloads that don't have that parameter.

IIIuminator commented:
I'll also add extensions for the extra options like `setBackground` that aren't inherited from `AlertDialog.Builder`

IIIuminator commented:
And add it to the material fun pack

IIIuminator commented:
> I'll also add extensions for the extra options like `setBackground` that aren't inherited from `AlertDialog.Builder`

nevermind, forgot that thats available through kotlins property access syntax

IIIuminator commented:
All done, I'll update `material-components` to `beta01` once its out and promote this to a full PR afterwards.

ivoberger commented:
beta-01 of material components is out now. I checked out the upstream develop branch and added the materialdialogs module and changed the build.gradle.kts in line with the other build files.
It's not building for me though. Seems to be missing implementation of some test classes

LouisCAD commented:
Material Components 1.1.0 has finally been released!
I'll merge this very soon, and release Splitties 3.0.0 after the changelog and doc updates are complete.

ivoberger commented:
👍 
Do you need me to do something here? Like fixing the merge conflicts or would you rather do that yourself?

LouisCAD commented:
For now I don't, but I'll let you know if that changes. The merge conflict is about a change in dependencies versions management, and some parts there are still WIP.

LouisCAD commented:
I'm sorry it took this long to merge this, but it's finally done! I made a few naming edits. It will be called "Alert Dialog Material" to be consistent with the AppCompat version and not overlap name of dialogs that would not be alert dialogs.
